proto/prehog/v1alpha/teleport.proto

```diff
@@ -268,8 +268,6 @@ enum DiscoverResource {
   DISCOVER_RESOURCE_DOC_DATABASE_RDS_PROXY = 34;
   DISCOVER_RESOURCE_DOC_DATABASE_HIGH_AVAILABILITY = 35;
   DISCOVER_RESOURCE_DOC_DATABASE_DYNAMIC_REGISTRATION = 36;
-
-  DISCOVER_RESOURCE_SAML_APPLICATION = 37;
```

Comment:
Once https://github.com/gravitational/cloud/pull/4972 is merged, I can re-update this proto and this field won't be removed.

Reply:
I'm bit confused. This field was added in https://github.com/gravitational/teleport/pull/27518/files#diff-53145009facaad79b5bba3127f8e08dad99c90dda4434c6b927f1d4100f2a2b3R272 and was backported to v13. Why we need gravitational/cloud#4972 ?

Reply:
I'm a little unclear as to which repo holds the source of truth from these protos, but it seems to make sense to me that this would be the cloud repo since that's where the server-side implementation is. If `teleport` is the source of truth, I can add this line back in.

Reply:
> The protobuf definitions are in github.com/gravitational/cloud/build/prehog/proto/prehog and are manually synchronized with the ones in github.com/gravitational/teleport/proto/prehog

The `manually synchronized` probably means copy and paste and since the gravitational/cloud#4972 is already merged I think that we should keep the  `DISCOVER_RESOURCE_SAML_APPLICATION` def.